tests/orchestrations_test.go

```diff
@@ -254,6 +254,48 @@ func Test_ActivityChain(t *testing.T) {
 	)
 }
 
+func Test_ActivityRetries(t *testing.T) {
+	// Registration
+	r := task.NewTaskRegistry()
+	r.AddOrchestratorN("ActivityRetries", func(ctx *task.OrchestrationContext) (any, error) {
+		if err := ctx.CallActivity("FailActivity", task.WithRetryPolicy(&task.ActivityRetryPolicy{
+			MaxAttempts:          3,
+			InitialRetryInterval: 10 * time.Millisecond,
+		})).Await(nil); err != nil {
+			return nil, err
+		}
+		return nil, nil
+	})
+	r.AddActivityN("FailActivity", func(ctx task.ActivityContext) (any, error) {
+		return nil, errors.New("activity failure")
+	})
+
+	// Initialization
+	ctx := context.Background()
+	exporter := initTracing()
+	client, worker := initTaskHubWorker(ctx, r)
+	defer worker.Shutdown(ctx)
+
+	// Run the orchestration
+	id, err := client.ScheduleNewOrchestration(ctx, "ActivityRetries")
+	if assert.NoError(t, err) {
+		metadata, err := client.WaitForOrchestrationCompletion(ctx, id)
+		if assert.NoError(t, err) {
+			assert.Equal(t, protos.OrchestrationStatus_ORCHESTRATION_STATUS_FAILED, metadata.RuntimeStatus)
```

Comment:
Please add an assert to validate that the completion time is at least 30ms after the start time, proving that the configured `InitialRetryInterval` is being honored.

Reply:
it should actually be 20 milliseconds I think, because only two timers will be created. On the last try we dont create a timer after it.